### PR TITLE
MB ideas for 2021

### PR DESCRIPTION
**Pull request by walterbender**
I've added a few Music Blocks ideas for 2021.

### CODE REVIEW

File: Ideas-2021.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+## Music Blocks 2.0 Block Graphics Refactoring
+
+Mentor: [Walter Bender](https://github.com/walterbender),
+Backup mentor: [Anindya Kundu](https://github.com/meganindya).
+
+**Prerequisites**<br>
+ - Experience with TypeScript/JavaScript
+ - Experience with graphic design
+ - Strong experience with SVG
+
```
Comment (meganindya) [inline Ideas-2021.md:21]: React v17 (with hooks) too?

File: Ideas-2021.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+**Description**<br> We are refactoring Music Blocks. This gives us an opportunity to revisit a number of UX issues, including the design and implementation of the block graphics. One thing that will not change is that we will use SVG for the block graphics, but other than that, we have the opportunity to rethink bot how blocks look (in particular, how they interlock and how they expand/resize) as well as the corresponding code we use to generate the block artwork SVG. Finally, it would be great to use CSS to define many of the block attributes, e.g., color and other styling features, rather than having everything hardcoded in the SVG. (Modern SVG supports this.)
+
+**Project Task Checklist**<br>
+ - Familiarize yourself with the current implementation [blockfactory](https://github.com/sugarlabs/musicblocks/blob/master/js/blockfactory.js)
+ - Come up with a framework for how the block interconnections will work -- we'd like to make it more obvious how blocks interlock and also use the interlocking to help define the schemas associated with some block logic.
+ - Design the class structure for the new block rendering approach.
+ - Implement all of the above in TypeScript.
```
Comment (meganindya) [inline Ideas-2021.md:28]: React (TypeScript)

File: Ideas-2021.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+## Music Blocks 2.0 Menus and Palettes
+Mentor: [Anindya Kundu](https://github.com/meganindya),
+Backup mentor: [Walter Bender](https://github.com/walterbender).
+
+**Prerequisites**<br>
+ - Experience with TypeScript/JavaScript
+ - Experience with user interface design
+ - Strong experience with Materials
+
```
Comment (meganindya) [inline Ideas-2021.md:47]: React like previous one

File: Ideas-2021.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+**Description**<br> We are refactoring Music Blocks. This gives us an opportunity to revisit a number of UX issues, including the design and implementation of the menu bars and block palettes. We have some sketches as to what it might look like, but your ideas are most welcome.
+
+**Project Task Checklist**<br>
+ - Familiarize yourself with the current implementations [toolbars](https://github.com/sugarlabs/musicblocks/blob/master/js/toolbar.js) [palettes](https://github.com/sugarlabs/musicblocks/blob/master/js/palette.js)
+ - Familiarize yourself with the M 2.0 sketches [wireframe](https://github.com/sugarlabs/musicblocks-2/discussions/5)
+ - Come up with a framework for the new toolbars and palettes
+ - Design the class structure for the toolbars
+ - Design the class structure for the palettes
+ - Implement all of the above in TypeScript.
```
Comment (meganindya) [inline Ideas-2021.md:56]: React(TypeScript)

**Review by meganindya — COMMENTED** [review]
Let's make it React 17 with TypeScript for the UI like we discussed?

**Comment by walterbender** [comment]
I managed to mess up the rebase...

**Pull request closed without merging**